astroquery/ipac/irsa/core.py

```diff
@@ -164,299 +96,88 @@ def query_region_async(self, coordinates=None, *, catalog=None,
+                if isinstance(radius, str):
+                    radius = Angle(radius)
+                where = (" WHERE CONTAINS(POINT('ICRS',ra,dec),"
+                         f"CIRCLE('ICRS',{ra},{dec},{radius.to(u.deg).value}))=1")
+            elif spatial == 'Box':
+                if isinstance(width, str):
+                    width = Angle(width)
+                where = (" WHERE CONTAINS(POINT('ICRS',ra,dec),"
+                         f"BOX('ICRS',{ra},{dec},{width.to(u.deg).value},{width.to(u.deg).value}))=1")
+            else:
+                raise ValueError("Unrecognized spatial query type. Must be one of "
+                                 "'Cone', 'Box', 'Polygon', or 'All-Sky'.")
 
-        # Convert to astropy.table.Table instance
-        table = first_table.to_table()
+        adql += where
 
-        # Check if table is empty
-        if len(table) == 0:
-            warnings.warn("Query returned no results, so the table will "
-                          "be empty", NoResultsWarning)
+        if get_query_payload:
+            return adql
+        response = self.query_tap(query=adql)
 
-        return table
+        return response.to_table()
 
-    def list_catalogs(self, cache=False):
+    @deprecated_renamed_argument("cache", None, since="0.4.7")
+    def list_catalogs(self, full=False, cache=False):
         """
-        Return a dictionary of the catalogs in the IRSA Gator tool.
+        Return information of available IRSA catalogs.
 
         Parameters
         ----------
-        cache : bool
-            Use local cache when set to `True`. Default is `False`.
-
-        Returns
-        -------
-        catalogs : dict
-            A dictionary of catalogs where the key indicates the catalog
-            name to be used in query functions, and the value is the verbose
-            description of the catalog.
+        full : bool
+            If True returns the full schema VOTable. If False returns a dictionary of the table names and
+            their description.
 
         """
-        response = self._request("GET", url=Irsa.GATOR_LIST_URL,
-                                 params=dict(mode='xml'), cache=cache,
-                                 timeout=Irsa.TIMEOUT)
+        tap_tables = Irsa.query_tap("SELECT * FROM TAP_SCHEMA.tables")
 
-        root = tree.fromstring(response.content)
-        catalogs = {}
-        for catalog in root.findall('catalog'):
-            catname = catalog.find('catname').text
-            desc = catalog.find('desc').text
-            catalogs[catname] = desc
-
-        return catalogs
+        if full:
+            return tap_tables
+        else:
+            return {tap_table['table_name']: tap_table['description'] for tap_table in tap_tables}
 
-    def print_catalogs(self, cache=False):
-        """
-        Display a table of the catalogs in the IRSA Gator tool.
-        """
-        catalogs = self.list_catalogs(cache=cache)
+    # TODO, deprecate this as legacy
```

Comment:
Is there a replacement for this, or are there just too many catalogs now to print all of them?

Reply:
This should still work, but I didn't really find it useful when I was going through the original code as there are hundreds of catalogs.